src/core/db/manager.py

```diff
@@ -259,13 +301,22 @@ def list_collaborators(self, repo):
         # [{'username': 'foo_user', 'permissions': 'UC'},
            {'username': 'bar_user', 'permissions': 'U'}]
         """
+        # get the database's idea of permissions
         with _superuser_connection(self.repo_base) as conn:
-            result = conn.list_collaborators(repo=repo)
-        return result
+            db_collabs = conn.list_collaborators(repo=repo)
+
+        # merge it with the datahub collaborator model permissions
+        dh_collabs = Collaborator.objects.all()
```

Comment:
This is grabbing all Collaborator objects and using the last to have the same username, regardless of whether the repo matches. Also, it keeps iterating after it finds a match.

Could be replaced with: 

```py
    usernames = (db_collab['username'] for db_collab in db_collabs)
    dh_collabs = Collaborator.objects.filter(user__username__in=usernames,
                                             repo_base=self.repo_base,
                                             repo_name=repo)
    for db_collab in db_collabs:
        db_collab['file_permissions'] = next(
            (dh_collab.file_permission for dh_collab in dh_collabs
                if dh_collab.user.username == db_collab['username']),
            '')
```

Note that this proposed snippet sets `db_collab['file_permissions'] == ''` if there's no matching dh_collab object. I'm not sure if that's desired behavior.